server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/node/Tree.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.compositeindex.datacube.startree.node;
+
+/**
+ * Interface for star-tree.
+ *
+ * @opensearch.experimental
+ */
+public interface Tree {
```

Comment:
If Tree is generic Node also must be generic. Tree cannot contain StarTreeNode. 

We can remove the abstraction as well , each structure might be quite different from each other.